svm/src/transaction_processor.rs

```diff
@@ -465,8 +473,18 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             fee_details.total_fee(),
         )?;
 
-        // Capture fee-subtracted fee payer account and original nonce account state
-        // to rollback to if transaction execution fails.
+        if let Some(ref mut nonce_info) = nonce {
+            // Coding instruction 0 is appropriate because it must be the advance nonce instruction
+            // However the nonce info is guaranteed to be valid so this should never occur
+            nonce_info
+                .try_advance_nonce(*durable_nonce, lamports_per_signature)
+                .map_err(|_| {
+                    TransactionError::InstructionError(0, InstructionError::InvalidAccountData)
+                })?;
+        };
```

Comment:
What do you think about moving the nonce advancing into `load_message_nonce_account` in the `check_transactions` module? That way, we can avoid the extra deserialization inside `try_advance_nonce`, we can the `next_durable_nonce` value there, and also error handling gets a little cleaner because then we just drop the transaction earlier.

Reply:
oh i think that would work yea, it doesnt looks like nonce info on the tx detail is used for anything other than checking if it exists

Reply:
Cool, can you give it a shot in this PR?

Reply:
done!